docs/src/best-practices-js.md

```diff
@@ -0,0 +1,171 @@
+---
+id: best-practices
+title: "Best Practices"
+---
+
+
+This guide should help you to make sure you are following our best practices and writing tests that are more resilient.
+## Use locators
+
+In order to write end to end tests we need to first find elements on the webpage. Automated tests should verify that the application code works for the end users. Implementation details are things which users of your code will not typically use, see, or even know about such as the name of a function or if it's an array. The end user will see or interact with what we render so our test should typically only see/interact with the rendered output.
```

Comment:
I like this first paragraph, however it doesn't seem to fit the locators section. I'd recommend add a bit of "test isolation" here, and make a separate first section about the "testing philosophy" or something like that 😄 : 1. test user-visible behavior, 2. make tests as isolated as possible.